sources/packages/web/src/views/student/StudentDashboard.vue

```diff
@@ -25,15 +45,43 @@
 import { computed } from "vue";
 import { useStore } from "vuex";
 import { StudentRoutesConst } from "../../constants/routes/RouteConstants";
-
+import { StudentService } from "@/services/StudentService";
+import { SetupContext, onMounted, ref } from "vue";
 export default {
-  setup() {
+  emits: ["update:formName", "change"],
+  props: {
+    formName: {
+      type: String,
+    },
+  },
+  setup(props: any, context: SetupContext) {
+    const programYearList = ref();
+    const onYearChange = (event: any) => {
+      context.emit("update:formName", event.value);
+      context.emit("change", event);
+    };
+    onMounted(async () => {
+      const programYears = await StudentService.shared.getProgramYears();
+      programYearList.value = programYears
+        ? programYears?.map((el: any) => ({
+            name: "(" + el.programYear + ") - " + el.programYearDesc,
+            code: el.formName,
+          }))
+        : [];
+    });
     const store = useStore();
     const user = computed(() => store.state.student.profile);
 
     return {
-      user,
       StudentRoutesConst,
+      programYearList,
+      user,
+      onYearChange,
+    };
+  },
+  data(props: any) {
```

Comment:
Can you please elaborate on why it is needed or why we cannot resolve it inside the setup method?